### ISSUE TITLE
optimize: use SIMD to accelerate json.Compact 

### ISSUE THREAD

**Issue by liuq19** [bytedance/sonic#414]
[Compat](https://github.com/bytedance/sonic/blob/ebbe7589ca6e2e39e915da3295a22b1a754f1871/encoder/utils.go#L36): now is use the encoding/json API, it is slow and we can use SIMD in native C to accelerate it.

**Comment by ngqinzhe** [comment]
Hi @liuq19, interested in working on this issue. I'm assuming [this](https://lemire.me/blog/2017/01/20/how-quickly-can-you-remove-spaces-from-a-string/) is relevant to what you want to achieve here. If this is the correct direction, I will start working on it.

**Comment by liuq19** [comment]
Yeah. I think it is a good solution, welcome to try it. And we could use shuffle to compute the space mask. https://github.com/bytedance/sonic/blob/main/native/fastbytes.c#L36

**Comment by liuq19** [comment]
hello, is there any update?
